tests/system/test_shutdown_message/test_shutdown_message.py

```diff
@@ -0,0 +1,95 @@
+'''
+copyright: Copyright (C) 2015-2023, Wazuh Inc.
+           Created by Wazuh, Inc. <info@wazuh.com>.
+           This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+type: system
+brief: sys
+tier: 0
+modules:
+    - enrollment
+components:
+    - manager
+    - agent
+daemons:
+    - wazuh-authd
+    - wazuh-agentd
+os_platform:
+    - linux
+os_version:
+    - Debian Buster
+references:
+    - https://documentation.wazuh.com/current/user-manual/registering/agent-enrollment.html
+'''
+
+import os
+import pytest
+import re
+import time
+from wazuh_testing import T_1, T_3
+from wazuh_testing.tools import WAZUH_PATH
+from wazuh_testing.tools.system import HostManager
+from system import restart_cluster
+
+inventory_path = os.path.join(os.path.dirname(os.path.dirname(os.path.abspath(__file__))),
+                              'provisioning', 'big_cluster_40_agents', 'inventory.yml')
+host_manager = HostManager(inventory_path)
+local_path = os.path.dirname(os.path.abspath(__file__))
+agent_conf_file = os.path.join(os.path.dirname(os.path.realpath(__file__)), '..',
+                               'provisioning', 'big_cluster_40_agents', 'roles', 'agent-role', 'files', 'ossec.conf')
+
+testinfra_hosts = ['wazuh-master', 'wazuh-worker1', 'wazuh-worker2']
+workers = ['wazuh-worker1', 'wazuh-worker2']
+agents = []
+number_agents = 40
+
+pytestmark = [pytest.mark.cluster, pytest.mark.big_cluster_40_agents_env]
+
+
+@pytest.fixture()
+def restart_all_agents():
+    for number_agent in range(number_agents):
+        agents.append(f'wazuh-agent{number_agent+1}')
+
+    restart_cluster(testinfra_hosts + agents, host_manager)
+
+    time.sleep(T_1)
+
+    yield
+
+    for agent in agents:
+        host_manager.run_command(agent, f'{WAZUH_PATH}/bin/wazuh-control restart')
+
+
+@pytest.fixture()
+def stop_gracefully_all_agents():
+    for agent in agents:
+        host_manager.run_command(agent, f'{WAZUH_PATH}/bin/wazuh-control stop')
+
+
+def test_shut_down_message_gracefully_stopped_agent(restart_all_agents, stop_gracefully_all_agents):
+    '''
+        description: Checking shutdown message when socket is closed.
+        wazuh_min_version: 4.5.0
```

Comment:
Wazuh min version is 4.5.0 or 4.6.0?

Reply:
Changed in https://github.com/wazuh/wazuh-qa/pull/4146/commits/d0592899d807833147849678e691e3da5974918c